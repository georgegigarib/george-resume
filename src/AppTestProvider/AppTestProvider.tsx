import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
