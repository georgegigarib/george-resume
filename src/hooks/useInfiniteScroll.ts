import { useEffect, useRef } from 'react'

const useInfiniteScroll = (containerId: string) => {
  const containerRef = useRef<HTMLDivElement>(null)

  useEffect(() => {
    const container = containerRef.current
    if (!container) return

    let animationFrame: number
    let scrollPosition = 0

    try {
      // Duplicamos el contenido para permitir un scroll infinito
      const originalContent = container.innerHTML
      container.innerHTML += originalContent

      // Calculamos el ancho total para ajustar el desplazamiento
      const scrollWidth = container.scrollWidth / 2

      const scrollStep = () => {
        scrollPosition += 0.5 // Ajusta la velocidad del desplazamiento aquÃ­
        if (scrollPosition >= scrollWidth) {
          scrollPosition = 0
        }
        container.scrollLeft = scrollPosition
        animationFrame = requestAnimationFrame(scrollStep)
      }

      scrollStep()
    } catch (error) {
      console.error('Error in useInfiniteScroll:', error)
    }

    // Limpieza al desmontar
    return () => {
      if (animationFrame) cancelAnimationFrame(animationFrame)
    }
  }, [containerId])

  return containerRef
}

export default useInfiniteScroll
