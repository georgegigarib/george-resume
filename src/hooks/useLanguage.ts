import { useTranslation } from 'react-i18next';

export const useLanguage = () => {
  const { i18n } = useTranslation();

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
    localStorage.setItem('language', lng);
  };

  const loadLanguageFromStorage = () => {
    const savedLanguage = localStorage.getItem('language') || 'en';
    i18n.changeLanguage(savedLanguage);
  };

  return { changeLanguage, loadLanguageFromStorage };
};
