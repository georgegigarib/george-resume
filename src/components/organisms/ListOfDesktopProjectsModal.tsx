import ListOfDesktopProjectsPreModal from '@/components/molecules/ListOfDesktopProjectsPreModal'
import { useTranslation } from 'react-i18next'
import { useIsMobile } from '@/hooks/useIsMobile'
import { Tooltip } from '@mui/material'
import clsx from 'clsx'
import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
import { useDispatch } from 'react-redux'
import { setProjectModalState } from '@/store/modal/modalStatus'

const ListOfDesktopProjectsModal = () => {
  const { t } = useTranslation()
  const isMobile = useIsMobile()
  const dispatch = useDispatch()

  const openModal = () => {
    dispatch(setProjectModalState(true))
  }

  return (
    <div
      className="rounded-xl     
    bg-app-lightModeBoxes-box1 
    dark:bg-app-darkModeBoxes-box1 
    dark:border-app-lightModeBoxes-box1 
    border-app-darkModeBoxes-box1 flex flex-col-reverse
    border-[2px]"
    >
      <ListOfDesktopProjectsPreModal />

      <div>
        <div
          onClick={openModal}
          className="absolute p-1 rounded-lg cursor-pointer items-center w-[7.2rem] md:w-44 dark:hover:bg-white hover:bg-opacity-25 hover:bg-white dark:hover:bg-opacity-25
            bg-app-lightModeBoxes-box1 m-1
    dark:bg-app-darkModeBoxes-box1 md:mt-2 md:ml-2 border-app-darkModeBoxes-box1 dark:border-app-lightModeBoxes-box1
    "
        >
          <div className="grid grid-cols-1 md:grid-cols-4 items-center px-2 pt-2 md:pb-2">
            <div className="col-span-3 flex justify-between flex-col h-[7.5rem] md:h-auto">
              <h1
                className={clsx(
                  'font-bold flex flex-wrap text-start dark:text-white text-black',
                  isMobile ? 'text-sm' : 'text-md'
                )}
              >
                {t('dev.projects.projects.desktop.title')}
              </h1>
              <Tooltip title={t('dev.projects.showAllProjects')} arrow>
                <div className="flex p-1 rounded-lg md:hidden">
                  <p className="mx-auto text-xs font-bold">{t('dev.projects.seeMore')}</p>
                </div>
              </Tooltip>
            </div>

            <div className="hidden md:flex justify-center ml-6">
              <Tooltip title={t('dev.projects.seeMore')} arrow>
                <ArrowForwardIcon
                  fontSize="large"
                  className="hover:bg-white hover:bg-opacity-25 rounded-full p-1 dark:text-white text-black"
                />
              </Tooltip>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default ListOfDesktopProjectsModal
