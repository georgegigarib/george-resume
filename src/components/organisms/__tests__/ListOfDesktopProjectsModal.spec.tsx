import { render, screen } from '@testing-library/react'
import { describe, expect, it, vi, beforeEach, Mock } from 'vitest'
import ListOfDesktopProjectsModal from '@/components/organisms/ListOfDesktopProjectsModal'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'
import { useIsMobile } from '@/hooks/useIsMobile'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: vi.fn(),
}))

vi.mock('@/constants/projects', () => ({
  desktopProjects: [
    {
      title: 'dev.projects.projects.desktop.exampleProject.title',
      description: 'dev.projects.projects.desktop.exampleProject.description',
      url: 'http://example.com/desktop1',
      tools: ['React', 'TypeScript'],
      src: 'desktop1.png',
    },
    {
      title: 'dev.projects.projects.desktop.exampleProject.title',
      description: 'dev.projects.projects.desktop.exampleProject.description',
      url: 'http://example.com/desktop2',
      tools: ['React', 'TypeScript'],
      src: 'desktop2.png',
    },
    {
      title: 'dev.projects.projects.desktop.exampleProject.title',
      description: 'dev.projects.projects.desktop.exampleProject.description',
      url: 'http://example.com/desktop3',
      tools: ['React', 'TypeScript'],
      src: 'desktop3.png',
    },
  ],
}))

describe('ListOfDesktopProjectsModal component', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  const renderComponent = (isMobile: boolean) => {
    ;(useIsMobile as Mock).mockReturnValue(isMobile)
    return render(
      <AppTestProvider>
        <ListOfDesktopProjectsModal />
      </AppTestProvider>
    )
  }

  it('matches the snapshot for desktop view', () => {
    const { asFragment } = renderComponent(false)
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches the snapshot for mobile view', () => {
    const { asFragment } = renderComponent(true)

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the correct content for desktop view', () => {
    renderComponent(false)

    expect(screen.getByText('Example desktop project')).toBeInTheDocument()
  })

  it('renders the correct content for mobile view', () => {
    renderComponent(true)
    expect(screen.getByText('Example desktop project')).toBeInTheDocument()
  })
})
