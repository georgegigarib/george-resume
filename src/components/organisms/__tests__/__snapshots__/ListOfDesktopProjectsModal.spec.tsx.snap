// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListOfDesktopProjectsModal component > matches the snapshot for desktop view 1`] = `
<DocumentFragment>
  <div
    class="rounded-xl     
    bg-app-lightModeBoxes-box1 
    dark:bg-app-darkModeBoxes-box1 
    dark:border-app-lightModeBoxes-box1 
    border-app-darkModeBoxes-box1 flex flex-col-reverse
    border-[2px]"
  >
    <div
      class="grid ml-8 grid-rows-1 grid-cols-4 
    bg-app-lightModeBoxes-box1 
    dark:bg-app-darkModeBoxes-box1 
     border-l-0 rounded-l-none
    rounded-xl"
    >
      <div
        class="text-center mb-4 col-span-1 self-center p-2 mr-4 
        md:dark:bg-app-lightModeBoxes-box1 
    md:bg-white
    md:bg-opacity-20
    md:border-[2px]
    border-app-darkModeBoxes-box1
    dark:border-white
    dark:border-opacity-60
    md:shadow-md
    opacity-90 rounded-xl flex flex-col gap-4 mt-16"
      >
        <h3
          class="hidden md:flex text-lg font-bold dark:text-white text-black"
        >
          Example desktop project
        </h3>
        <div
          class="hidden md:flex flex-wrap justify-center gap-2"
        >
          <i
            aria-label="React"
            class="ci ci-React ci-lg"
            data-mui-internal-clone-element="true"
          />
          <i
            aria-label="TypeScript"
            class="ci ci-TypeScript ci-lg"
            data-mui-internal-clone-element="true"
          />
        </div>
      </div>
      <div
        class="col-span-3 self-center py-3 pr-3 ml-3 group"
      >
        <div
          class="swiper swiper-initialized swiper-vertical h-[120px] md:h-[250px] w-full"
        >
          <div
            class="swiper-wrapper"
            style="cursor: grab; transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(0px, NaNpx, 0px);"
          >
            <div
              class="swiper-slide bg-transparent"
              data-swiper-slide-index="0"
            >
              <img
                alt="dev.projects.projects.desktop.exampleProject.title"
                class="rounded-2xl"
                src="desktop1.png"
              />
            </div>
            <div
              class="swiper-slide bg-transparent"
              data-swiper-slide-index="1"
            >
              <img
                alt="dev.projects.projects.desktop.exampleProject.title"
                class="rounded-2xl"
                src="desktop2.png"
              />
            </div>
            <div
              class="swiper-slide bg-transparent"
              data-swiper-slide-index="2"
            >
              <img
                alt="dev.projects.projects.desktop.exampleProject.title"
                class="rounded-2xl"
                src="desktop3.png"
              />
            </div>
          </div>
          <div
            class="swiper-pagination swiper-pagination-bullets swiper-pagination-vertical"
          >
            <span
              class="swiper-pagination-bullet"
            />
            <span
              class="swiper-pagination-bullet"
            />
            <span
              class="swiper-pagination-bullet swiper-pagination-bullet-active"
            />
          </div>
          <div
            class="hidden md:block"
          >
            <div
              class="autoplay-progress"
              slot="container-end"
            >
              <svg
                style="--progress: 0;"
                viewBox="0 0 48 48"
              >
                <circle
                  cx="24"
                  cy="24"
                  r="20"
                />
              </svg>
              <span>
                3s
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="absolute p-1 rounded-lg cursor-pointer items-center w-[7.2rem] md:w-44 dark:hover:bg-white hover:bg-opacity-25 hover:bg-white dark:hover:bg-opacity-25
            bg-app-lightModeBoxes-box1 m-1
    dark:bg-app-darkModeBoxes-box1 md:mt-2 md:ml-2 border-app-darkModeBoxes-box1 dark:border-app-lightModeBoxes-box1
    "
      >
        <div
          class="grid grid-cols-1 md:grid-cols-4 items-center px-2 pt-2 md:pb-2"
        >
          <div
            class="col-span-3 flex justify-between flex-col h-[7.5rem] md:h-auto"
          >
            <h1
              class="font-bold flex flex-wrap text-start dark:text-white text-black text-md"
            >
              Self taste projects üñ•Ô∏è
            </h1>
            <div
              aria-label="Show all projects"
              class="flex p-1 rounded-lg md:hidden"
              data-mui-internal-clone-element="true"
            >
              <p
                class="mx-auto text-xs font-bold"
              >
                See more
              </p>
            </div>
          </div>
          <div
            class="hidden md:flex justify-center ml-6"
          >
            <svg
              aria-hidden="true"
              aria-label="See more"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge hover:bg-white hover:bg-opacity-25 rounded-full p-1 dark:text-white text-black css-19vn4a9-MuiSvgIcon-root"
              data-mui-internal-clone-element="true"
              data-testid="ArrowForwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListOfDesktopProjectsModal component > matches the snapshot for mobile view 1`] = `
<DocumentFragment>
  <div
    class="rounded-xl     
    bg-app-lightModeBoxes-box1 
    dark:bg-app-darkModeBoxes-box1 
    dark:border-app-lightModeBoxes-box1 
    border-app-darkModeBoxes-box1 flex flex-col-reverse
    border-[2px]"
  >
    <div
      class="grid ml-8 grid-rows-1 grid-cols-4 
    bg-app-lightModeBoxes-box1 
    dark:bg-app-darkModeBoxes-box1 
     border-l-0 rounded-l-none
    rounded-xl"
    >
      <div
        class="text-center mb-4 col-span-1 self-center p-2 mr-4 
        md:dark:bg-app-lightModeBoxes-box1 
    md:bg-white
    md:bg-opacity-20
    md:border-[2px]
    border-app-darkModeBoxes-box1
    dark:border-white
    dark:border-opacity-60
    md:shadow-md
    opacity-90 rounded-xl flex flex-col gap-4 mt-16"
      >
        <h3
          class="hidden md:flex text-lg font-bold dark:text-white text-black"
        >
          Example desktop project
        </h3>
        <div
          class="hidden md:flex flex-wrap justify-center gap-2"
        >
          <i
            aria-label="React"
            class="ci ci-React ci-lg"
            data-mui-internal-clone-element="true"
          />
          <i
            aria-label="TypeScript"
            class="ci ci-TypeScript ci-lg"
            data-mui-internal-clone-element="true"
          />
        </div>
      </div>
      <div
        class="col-span-3 self-center py-3 pr-3 ml-3 group"
      >
        <div
          class="swiper swiper-initialized swiper-vertical h-[120px] md:h-[250px] w-full"
        >
          <div
            class="swiper-wrapper"
            style="cursor: grab; transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(0px, NaNpx, 0px);"
          >
            <div
              class="swiper-slide bg-transparent"
              data-swiper-slide-index="0"
            >
              <img
                alt="dev.projects.projects.desktop.exampleProject.title"
                class="rounded-2xl"
                src="desktop1.png"
              />
            </div>
            <div
              class="swiper-slide bg-transparent"
              data-swiper-slide-index="1"
            >
              <img
                alt="dev.projects.projects.desktop.exampleProject.title"
                class="rounded-2xl"
                src="desktop2.png"
              />
            </div>
            <div
              class="swiper-slide bg-transparent"
              data-swiper-slide-index="2"
            >
              <img
                alt="dev.projects.projects.desktop.exampleProject.title"
                class="rounded-2xl"
                src="desktop3.png"
              />
            </div>
          </div>
          <div
            class="swiper-pagination swiper-pagination-bullets swiper-pagination-vertical"
          >
            <span
              class="swiper-pagination-bullet"
            />
            <span
              class="swiper-pagination-bullet"
            />
            <span
              class="swiper-pagination-bullet swiper-pagination-bullet-active"
            />
          </div>
          <div
            class="hidden md:block"
          >
            <div
              class="autoplay-progress"
              slot="container-end"
            >
              <svg
                style="--progress: 0;"
                viewBox="0 0 48 48"
              >
                <circle
                  cx="24"
                  cy="24"
                  r="20"
                />
              </svg>
              <span>
                3s
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="absolute p-1 rounded-lg cursor-pointer items-center w-[7.2rem] md:w-44 dark:hover:bg-white hover:bg-opacity-25 hover:bg-white dark:hover:bg-opacity-25
            bg-app-lightModeBoxes-box1 m-1
    dark:bg-app-darkModeBoxes-box1 md:mt-2 md:ml-2 border-app-darkModeBoxes-box1 dark:border-app-lightModeBoxes-box1
    "
      >
        <div
          class="grid grid-cols-1 md:grid-cols-4 items-center px-2 pt-2 md:pb-2"
        >
          <div
            class="col-span-3 flex justify-between flex-col h-[7.5rem] md:h-auto"
          >
            <h1
              class="font-bold flex flex-wrap text-start dark:text-white text-black text-sm"
            >
              Self taste projects üñ•Ô∏è
            </h1>
            <div
              aria-label="Show all projects"
              class="flex p-1 rounded-lg md:hidden"
              data-mui-internal-clone-element="true"
            >
              <p
                class="mx-auto text-xs font-bold"
              >
                See more
              </p>
            </div>
          </div>
          <div
            class="hidden md:flex justify-center ml-6"
          >
            <svg
              aria-hidden="true"
              aria-label="See more"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge hover:bg-white hover:bg-opacity-25 rounded-full p-1 dark:text-white text-black css-19vn4a9-MuiSvgIcon-root"
              data-mui-internal-clone-element="true"
              data-testid="ArrowForwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
